prov/tcp/src/tcpx.h

```diff
@@ -213,18 +213,23 @@ struct tcpx_fabric {
 	pthread_t		conn_mgr_thread;
 };
 
-struct tcpx_msg_data {
-	size_t		iov_cnt;
+struct tcpx_rma_data {
+	size_t			rma_iov_cnt;
 	union {
-		struct iovec		iov[TCPX_IOV_LIMIT+1];
-		struct fi_rma_iov	rma_iov[TCPX_IOV_LIMIT+1];
-		struct fi_rma_ioc	rma_ioc[TCPX_IOV_LIMIT+1];
+		struct fi_rma_iov	rma_iov[TCPX_IOV_LIMIT];
+		struct fi_rma_ioc	rma_ioc[TCPX_IOV_LIMIT];
 	};
+};
+
+struct tcpx_msg_data {
+	size_t		iov_cnt;
+	struct iovec		iov[TCPX_IOV_LIMIT+1];
 	uint8_t			inject[TCPX_MAX_INJECT_SZ];
 };
```

Comment:
The reason for separating msg_data from rma_data isn't obvious.  Please add more details to the commit messages!  Note that RMA writes can still be sent using inject.

Reply:
I want to have rma_data before I get the actual message data so that I can directly place it/read it from, at the remote side.  if i have a same structure holding both rma and msg iov, then it is not possible since i am using scatter-gather read call to get data. 

one way to optimize is to coalesce rma iov with msg hdr, thus decreasing number of iovs.